### PR TITLE
chore: update for mrk support in keystore

### PR DESCRIPTION
**Pull request by ajewellamz**
_Issue #, if available:_

_Description of changes:_

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Check any applicable:

- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

File: framework/branch-key-store.md [hunk 1]
```diff
@@ -89,6 +89,21 @@ The table name of the DynamoDb table that backs this Keystore.
 A valid [AWS KMS Key ARN](./aws-kms/aws-kms-key-arn.md#a-valid-aws-kms-arn)
 that wraps and unwraps keys stored in Amazon DynamoDB.
 
+The KMS Configuration MUST distinguish between strict ARN compatibility and MRK ARN compatibility.
+
+Both compatibility modes are allowed with both mrk ARN's and non-mrk ARNs.
```
Comment (josecorella) [inline framework/branch-key-store.md:94]:
```suggestion
Both compatibility modes are allowed with both mrk ARN's and single region ARNs.
```

File: framework/branch-key-store.md [hunk 2]
```diff
@@ -89,6 +89,21 @@ The table name of the DynamoDb table that backs this Keystore.
...
+
+#### AWS Key Compatibility
```
Comment (josecorella) [inline framework/branch-key-store.md:96]:
```suggestion
#### AWS Key ARN Compatibility
```

File: framework/branch-key-store.md [hunk 3]
```diff
@@ -89,6 +89,21 @@ The table name of the DynamoDb table that backs this Keystore.
...
+
+For two ARNs to be compatible :
+
+If the [AWS KMS Configuration](#aws-kms-configuration) designates strict ARN compatibility,
```
Comment (josecorella) [inline framework/branch-key-store.md:100]:
```suggestion
If the [AWS KMS Configuration](#aws-kms-configuration) designates single region ARN compatibility,
```

**Review by josecorella — COMMENTED** [review]
looks good!
before approving, let's get someone else to review and change the PR title to be compliant with semantic commits

File: framework/branch-key-store.md [hunk 4]
```diff
@@ -206,7 +221,7 @@ The operation MUST call [AWS KMS API GenerateDataKeyWithoutPlaintext](https://do
 The call to AWS KMS GenerateDataKeyWithoutPlaintext MUST use the configured AWS KMS client to make the call.
 The operation MUST call AWS KMS GenerateDataKeyWithoutPlaintext with a request constructed as follows:
 
-- `KeyId` MUST be the configured `AWS KMS Key ARN` in the [AWS KMS Configuration](#aws-kms-configuration) for this keystore
+- `KeyId` MUST be [compatible with](#aws-key-arn-compatibility) the configured `AWS KMS Key ARN` in the [AWS KMS Configuration](#aws-kms-configuration) for this keystore
```
Comment (rishav-karanjit) [inline framework/branch-key-store.md:224]: very minor nit:

```suggestion
- `KeyId` MUST be [compatible with](#aws-key-arn-compatibility) the configured `AWS KMS Key ARN` in the [AWS KMS Configuration](#aws-kms-configuration) for this keystore.
```

File: framework/branch-key-store.md [hunk 5]
```diff
@@ -222,7 +237,7 @@ The operation MUST call [AWS KMS API GenerateDataKeyWithoutPlaintext](https://do
...
```
Comment (rishav-karanjit) [inline framework/branch-key-store.md:240]:
```suggestion
- `KeyId` MUST be [compatible with](#aws-key-arn-compatibility) the configured `AWS KMS Key ARN` in the [AWS KMS Configuration](#aws-kms-configuration) for this keystore.
```

File: framework/branch-key-store.md [hunk 6]
```diff
@@ -235,10 +250,10 @@ The operation MUST call [AWS KMS API ReEncrypt](https://docs.aws.amazon.com/kms/
 with a request constructed as follows:
 
 - `SourceEncryptionContext` MUST be the [DECRYPT_ONLY encryption context for branch keys](#decrypt_only-encryption-context).
-- `SourceKeyId` MUST be the configured `AWS KMS Key ARN` in the [AWS KMS Configuration](#aws-kms-configuration) for this keystore
+- `SourceKeyId` MUST be [compatible with](#aws-key-arn-compatibility) the configured `AWS KMS Key ARN` in the [AWS KMS Configuration](#aws-kms-configuration) for this keystore
```
Comment (rishav-karanjit) [inline framework/branch-key-store.md:253]:
```suggestion
- `SourceKeyId` MUST be [compatible with](#aws-key-arn-compatibility) the configured `AWS KMS Key ARN` in the [AWS KMS Configuration](#aws-kms-configuration) for this keystore.
```

File: framework/branch-key-store.md [hunk 7]
```diff
@@ -235,10 +250,10 @@ The operation MUST call [AWS KMS API ReEncrypt](https://docs.aws.amazon.com/kms/
...
 - `CiphertextBlob` MUST be the wrapped DECRYPT_ONLY Branch Key.
 - ReEncrypt `GrantTokens` MUST be this keystore's [grant tokens](https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#grant_token).
-- `DestinationKeyId` MUST be the configured `AWS KMS Key ARN` in the [AWS KMS Configuration](#aws-kms-configuration) for this keystore
+- `DestinationKeyId` MUST be [compatible with](#aws-key-arn-compatibility) the configured `AWS KMS Key ARN` in the [AWS KMS Configuration](#aws-kms-configuration) for this keystore
```
Comment (rishav-karanjit) [inline framework/branch-key-store.md:256]:
```suggestion
- `DestinationKeyId` MUST be [compatible with](#aws-key-arn-compatibility) the configured `AWS KMS Key ARN` in the [AWS KMS Configuration](#aws-kms-configuration) for this keystore.
```

File: framework/branch-key-store.md [hunk 8]
```diff
@@ -89,6 +89,21 @@ The table name of the DynamoDb table that backs this Keystore.
 A valid [AWS KMS Key ARN](./aws-kms/aws-kms-key-arn.md#a-valid-aws-kms-arn)
 that wraps and unwraps keys stored in Amazon DynamoDB.
 
+The KMS Configuration MUST distinguish between single region ARN compatibility and MRK ARN compatibility.
+
+Both compatibility modes are allowed with both mrk ARN's and single region ARNs.
+
+#### AWS Key ARN Compatibility
+
+For two ARNs to be compatible :
```
Comment (rishav-karanjit) [inline framework/branch-key-store.md:98]:
```suggestion
For two ARNs to be compatible:
```

**Review by justplaz — APPROVED** [review]
LGTM

**Review by josecorella — APPROVED** [review]

**Comment by texastony** [comment]
@ajewellamz Did the "no admin actions" change get in here?
I do not see them.

**Pull request merged**